### PR TITLE
XL/objectLayer: Save additional meta data during PutObject

### LINKED ISSUE #1602: objectLayer: Save md5sum/content-type/content-encoding validated during PutObject

### PR DESCRIPTION
**Pull request by balamurugana**
Fixes  #1602

### CODE REVIEW

File: xl-objects.go [hunk 1]
```diff
@@ -380,6 +415,10 @@ func (xl xlObjects) deleteObject(bucket, object string) error {
 	if ok, err := isMultipartObject(xl.storage, bucket, object); err != nil {
 		return err
 	} else if !ok {
+		metaJSONFile := path.Join(object, "meta.json")
+		if err = xl.storage.DeleteFile(bucket, metaJSONFile); err != nil {
+			return err
+		}
 		if err = xl.storage.DeleteFile(bucket, object); err != nil {
 			return err
 		}
```
Comment (harshavardhana) [inline xl-objects.go:424]: Now this is done, GetObject, GetObjectInfo need to populate these values.  Did you test this with listing files properly?

Comment (balamurugana) [inline xl-objects.go:424]: I tested by below mc commands

``` bash
mc cp 4mbfile myminio/mybucket/zero
mc ls -r myminio/mybucket
```

They work fine.

**Comment by harshavardhana** [comment]
> I tested by below mc commands
> 
> mc cp 4mbfile myminio/mybucket/zero
> mc ls -r myminio/mybucket
> They work fine.

Wanted to see if you can copy like around 1000's of files and be able to list them non-recursively and recursively.

**Comment by harshavardhana** [comment]
Tested this is working for all cases because the erasure layer is properly filtering this one out.

**Comment by harshavardhana** [comment]
Next is we need to read this file and reply back in GetObject(), GetObjectInfo().

**Pull request merged**